File: secondary_implementations/python/modules/InstagramModule.py
# vim: ts=2 sw=2 noexpandtab
import requests
from .AbstractModule import AbstractModule


class InstagramModule(AbstractModule):
	def try_login(self, uname, password):
		agent = AbstractModule().user_agent() 
		headers = {'user-agent': agent}
		resp = requests.get('https://www.instagram.com/accounts/login/',
												headers= headers)

		payload = {'username': uname, 'password': password}
		headers = {
			'user-agent': agent,
			'Content-Type': 'application/x-www-form-urlencoded',
			'X-CSRFToken': resp.cookies['csrftoken'],
			'Accept': '*/*',
			'Referer': 'https://www.instagram.com'}
		loginResp = requests.post(
			'https://www.instagram.com/accounts/login/ajax/', data=payload,
			headers=headers, cookies=resp.cookies, allow_redirects=False)
		
		return loginResp.json()['authenticated'] == True
		

instagram = InstagramModule()

File: secondary_implementations/ruby/modules/InstagramModule.rb
require_relative 'AbstractModule'  

class InstagramModule < AbstractModule  
  def try_login(username, password)  
    return false #Fails with Cookie errors (403 Forbidden)
  	@user = user_agent
    headers = { 'user-agent' => @user}
  	response = RestClient.get("https://www.instagram.com/accounts/login/", :headers => headers){|response, request, result| response }

  	payload = {'username' => username, 'password' => password}
  	headers = { 'user-agent' => @user, 'Content-Type' => 'application/x-www-form-urlencoded', 'X-CSRFToken' => response.cookies['csrftoken'], "Accept" => "*/*", "Referer" => "https://www.instagram.com"}
  	response = RestClient.post("https://www.instagram.com/accounts/login/ajax/", payload, {:headers => headers, :cookies => response.cookies, :max_redirects => 0}){|response, request, result| response }
  	response = JSON.parse(response.body)

  	return response['authenticated'] == true
  end  
end  